test/integration/BPool.t.sol

```diff
@@ -92,20 +105,39 @@ abstract contract PoolSwapIntegrationTest is Test, GasSnapshot {
     vm.startPrank(lp);
 
     uint256 lpBalance = pool.balanceOf(address(lp));
-    pool.exitPool(lpBalance, new uint256[](2));
+    pool.exitPool(lpBalance, new uint256[](3));
 
     assertEq(dai.balanceOf(address(lp)), DAI_LP_AMOUNT - DAI_OUT_AMOUNT_INVERSE); // initial 4k - ~100 dai
     assertEq(weth.balanceOf(address(lp)), WETH_LP_AMOUNT + WETH_AMOUNT_INVERSE); // initial 1 + 0.1 eth
+    assertEq(wbtc.balanceOf(lp), WBTC_LP_AMOUNT); // initial 0.1 wbtc
+  }
+
+  function testSimpleJoin() public {
+    _makeJoin();
+    assertEq(dai.balanceOf(joiner.addr), 0);
+    assertEq(weth.balanceOf(joiner.addr), 0);
+    assertEq(wbtc.balanceOf(joiner.addr), 0);
+  }
+
+  function testSimpleExit() public {
+    _makeExit();
+    assertEq(dai.balanceOf(exiter.addr), DAI_LP_AMOUNT / 100);
+    assertEq(weth.balanceOf(exiter.addr), WETH_LP_AMOUNT / 100);
+    assertEq(wbtc.balanceOf(exiter.addr), WBTC_LP_AMOUNT / 100);
   }
 
   function _deployFactory() internal virtual returns (IBFactory);
 
   function _makeSwap() internal virtual;
 
   function _makeSwapInverse() internal virtual;
+
+  function _makeJoin() internal virtual;
```

Comment:
why bother adding them as virtual if they're only going to be implemented in `BPool` tests? we could inline the code on `BPoolIntegrationTests`

Reply:
i did it like this to follow the inheritable pattern (even tho we're not using it rn), wdyt? which one do you prefer? 

the thing is if we do it inline, we must declare the test function inside BPool but not inside BCoWPool (since it the add/rm liquidity flow is not implemented in CoW)

Reply:
I mean it _is_ implemented but doesn't differ from BPool so it doesn't make sense to test, right?

I think it's okay to keep them if this is going to be merged into mainline balancer, since this could be a good start to test other BPool extensions